tests/common_test.py

```diff
@@ -96,6 +102,60 @@ def fixed_point_fun(params, *args, **kwargs):
     params, state = update(params=None, state=state0)
     test_util.check_states_have_same_types(state0, state)
 
+  def test_n_calls(self):
+    global N_CALLS
+
+    class LeastSquaresNCalls(LeastSquares):
+      def __call__(self, W, data):
+        global N_CALLS
+        N_CALLS += 1
+        return super().__call__(W, data)
+
+    fun = LeastSquaresNCalls()
+    root_fun = jax.grad(fun)
+
+    def fixed_point_fun(params, *args, **kwargs):
+      return root_fun(params, *args, **kwargs) - params
+
+
+    X, y = datasets.make_classification(n_samples=10, n_features=5, n_classes=3,
+                                        n_informative=3, random_state=0)
+    data = (X, y)
+    params0 = jnp.zeros(X.shape[1])
+    common_kwargs = dict(jit=False, maxiter=30)
+    fpi = jaxopt.FixedPointIteration(fixed_point_fun=fixed_point_fun, **common_kwargs)
+
+    # Gradient solvers and fixed point
+    for solver in (jaxopt.PolyakSGD(fun=fun, **common_kwargs),
+                   fpi,
+                   jaxopt.Broyden(fun=root_fun, **common_kwargs),
+                   jaxopt.AndersonAcceleration(fixed_point_fun=fixed_point_fun, **common_kwargs),
+                   jaxopt.BFGS(fun, **common_kwargs),
+                   jaxopt.LBFGS(fun=fun, **common_kwargs)):
```

Comment:
Can you add more of them here?

Reply:
which ones were you thinking about? For the (prox/proj) gradient descent ones because of the current fista line search it is not possible to have the right results.

I added nonlinear CG

Reply:
(which allowed to correct a zoom LS related bug)

Reply:
All the ones for which you added support for `num_fun_evals`.

Reply:
Sure : added MirrorDescent, but it's quite a big one

For BlockCD I can't add it because BlockCD does not support non-jitted functions yet (so it's not possible to build the unit test with the global variable like I did). I added a fixme, and will report this in an issue (should be a similar resolution as what @vroulet did with the cond in the LS).